### ISSUE TITLE
Grid view for level select from Pancelor's fork

### ISSUE THREAD

**Issue by Draknek** [david-pfx/PuzzleScriptNext#82]
A level select feature that would be nice is to have the levels shown in a grid rather than a list, so you can fit more onscreen at once.

![image](https://github.com/user-attachments/assets/4ab64069-9550-4ab4-be20-3833286aa0fb)

For games with numbered levels rather than named levels, it's a bit cleaner.

https://pancelor.com/PuzzleScript/editor.html?hack=e7402e0b3aa51eaaaa860400edfc0263

This is Pancelor's fork: https://github.com/pancelor/puzzlescript

Note: in this branch, it uses the `enable_level_select` prelude flag.

**Comment by david-pfx** [comment]
Yes it would be a nice thing to have. However, having had a wander through the code, it's a long way from being just a drop-in. It seems to depend on several chunks of code scattered through 3 files, a couple of globals and a new state variable. 

The style is not something to be proud of IMO, and it's diverged quite a long way from PS+.

So I hope to get round to it, just not yet.
